### PR TITLE
Add publishedAt timestamp

### PR DESCRIPTION
**Pull request by matthiasthieroff**
Add a publishedAt field to dogu v1 and v2 struct.
The field is automatically set when a new dogu is created.

### CODE REVIEW

**Comment by robertauer** [comment]
Please update the changelog.

File: core/dogu_v2.go [hunk 1]
```diff
@@ -601,6 +602,16 @@ type Dogu struct {
...
+	// PublishedAt is the date and time when the dogu was created.
+  //
+  // This field does not need to be filled in by the developer. The dogu.cloudogu.com service automatically replaces
+  // the content with the current time when the new dogu is created.
+  //
+  // Examples:
+  //   - 2024-10-16T07:49:34.738Z
+  //   - 2019-05-03T13:31:48.612Z
+  //
+	PublishedAt time.Time
```
Comment (alexander-dammeier) [inline core/dogu_v2.go]: fix indentation

**Comment by alexander-dammeier** [comment]
Seems that there are multiple sections, where the indentation is not like in the rest of the code. Can you please fix that?

File: core/dogu_v1.go [hunk 2]
```diff
@@ -12,6 +13,7 @@ import (
 type DoguV1 struct {
 	Name                 string
 	Version              string
+	PublishedAt          time.Time
```
Comment (nfranzeck) [inline core/dogu_v1.go:16]: Do you want to operate on a zero value for time.Time? If so, marshaling will create json with initial values as you can see in your tests above. Another option is to use PublishedAt as a pointer together with the json tag:

`PublishedAt          time.Time \`json:",omitempty"\``  this way PublishedAt will not be written when it is empty.  

https://go.dev/play/p/6nqE7CmRJL2

Comment (matthiasthieroff) [reply core/dogu_v1.go:16]: We don't have a particular opinion on this and would change it if you prefer.
The field will actually always be set and imho a zero value can/should not occur.

File: core/dogu_v2.go (hunk 1 above)
Comment (matthiasthieroff) [reply core/dogu_v2.go]: Indentation should be fixed in the latest commit. There was a moment where prettier stopped working and proper formatting didn't happen on save.

**Review by nfranzeck — APPROVED** [review]
LGTM

**Pull request merged**